File: LeetCode/3379/3379.cpp
class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {
        std::vector<int> cloned = nums;
        int len = nums.size();
        for(int i{0}; i < len;++i)
        {
            cloned[i] = nums[(((i + nums[i]) % len) + len) % len];
        }
        return cloned;
    }
};

File: LeetCode/3379/3379.rs
impl Solution {
    pub fn construct_transformed_array(nums: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::with_capacity(nums.len());
        nums.iter().enumerate().for_each(|(index, &elem)| result.push(nums[((((index as i32 + elem ) % nums.len() as i32) + nums.len() as i32) % nums.len()as i32) as usize]));
        result
    }
}
